mixer/adapter/solarwinds/appoptics/batching_test.go

```diff
@@ -0,0 +1,185 @@
+// limitations under the License.
+
+package appoptics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"istio.io/istio/mixer/adapter/solarwinds/papertrail"
+)
+
+func TestBatchMeasurements(t *testing.T) {
+
+	t.Run("All Good", func(t *testing.T) {
+		logger := &papertrail.LoggerImpl{}
+		logger.Infof("Starting %s - test run. . .", t.Name())
+		defer logger.Infof("Finished %s - test run. . .", t.Name())
+		prepChan := make(chan []*Measurement)
+		pushChan := make(chan []*Measurement)
+		stopChan := make(chan struct{})
+
+		loopFactor := true
+
+		go BatchMeasurements(&loopFactor, prepChan, pushChan, stopChan, logger)
+
+		go func() {
+			measurements := []*Measurement{}
+			for i := 0; i < MeasurementPostMaxBatchSize+1; i++ {
+				measurements = append(measurements, &Measurement{})
+			}
+			prepChan <- measurements
+			loopFactor = false
+			time.Sleep(time.Millisecond)
+			close(prepChan)
+			close(pushChan)
+		}()
+		count := 0
+		for range pushChan {
+			count++
+		}
+		if count != 1 {
+			t.Errorf("Batching is not working properly. Expected batches is 1 but got %d", count)
+		}
+		close(stopChan)
+	})
+
+	t.Run("Using stop chan", func(t *testing.T) {
+		logger := &papertrail.LoggerImpl{}
+		logger.Infof("Starting %s - test run. . .", t.Name())
+		defer logger.Infof("Finished %s - test run. . .", t.Name())
+		prepChan := make(chan []*Measurement)
+		pushChan := make(chan []*Measurement)
+		stopChan := make(chan struct{})
+
+		loopFactor := true
+		go func() {
+			time.Sleep(time.Millisecond)
+			stopChan <- struct{}{}
+		}()
+		BatchMeasurements(&loopFactor, prepChan, pushChan, stopChan, logger)
+		loopFactor = false
+		close(prepChan)
+		close(pushChan)
+		close(stopChan)
+	})
+
+}
+
+type MockServiceAccessor struct {
+	// MeasurementsService implements an interface for dealing with  Measurements
+	MockMeasurementsService func() MeasurementsCommunicator
+}
+
+func (s *MockServiceAccessor) MeasurementsService() MeasurementsCommunicator {
+	return s.MockMeasurementsService()
+}
+
+func TestPersistBatches(t *testing.T) {
+	tests := []struct {
+		name           string
+		expectedCount  int64
+		response       *http.Response
+		error          error
+		sendOnStopChan bool
+	}{
+		{
+			name:          "Persist all good",
+			expectedCount: 0,
+			response: &http.Response{
+				Status:     http.StatusText(http.StatusOK),
+				StatusCode: http.StatusOK,
+			},
+			error:          nil,
+			sendOnStopChan: false,
+		},
+		{
+			name:           "Response error",
+			expectedCount:  1,
+			response:       nil,
+			error:          fmt.Errorf("damn"),
+			sendOnStopChan: false,
+		},
+		{
+			name:           "Stop chan test",
+			expectedCount:  0,
+			response:       nil,
+			error:          nil,
+			sendOnStopChan: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			logger := &papertrail.LoggerImpl{}
+			logger.Infof("Starting %s - test run. . .\n", t.Name())
```

Comment:
any reason not to use `t.Log()` here ?

Reply:
Mainly just for uniformity. I just switched to using env.Logger(), where env is adapter test.NewEnv(t). Will this be ok?